x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/restriction/WorkflowService.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.authz.restriction;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.common.util.concurrent.ThreadContext;
+import org.elasticsearch.license.XPackLicenseState;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.rest.RestHandler;
+import org.elasticsearch.xpack.core.security.authz.restriction.Workflow;
+import org.elasticsearch.xpack.core.security.authz.restriction.WorkflowResolver;
+
+import static org.elasticsearch.core.Strings.format;
+
+public class WorkflowService {
+
+    private static final Logger logger = LogManager.getLogger(WorkflowService.class);
+
+    public static final String WORKFLOW_HEADER = "_xpack_security_workflow";
+
+    private final XPackLicenseState licenseState;
```

Comment:
Now I think we might not need license check at all. The search application APIs already [enforce license checks](https://github.com/elastic/elasticsearch/blob/1cc1d124327fca6a638aac8656873336b64fee5a/x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/action/SearchApplicationTransportAction.java#L55). Unless we want to license the workflow feature separately (and I don't think that's the case), we can just proceed regardlessly here.

PS: The current license check for enterprise search APIs is done at transport layer. I think it's better to be moved earlier to REST layer. I have raised a question in the slack channel.

Reply:
Thanks for raising this. After internal discussions, we agreed that it is not necessary to add a license check here since workflows are simply allowing existing user-facing feature (Search Application) to work correctly. The Search Application has already implemented proper license checks, which make license check of workflows unnecessary and redundant.